dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumMonitor.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.android.rum
+
+import com.datadog.android.Datadog
+import com.datadog.android.core.internal.CoreFeature
+import com.datadog.android.core.internal.utils.devLogger
+import com.datadog.android.rum.internal.RumFeature
+import com.datadog.android.rum.internal.monitor.DatadogRumMonitor
+import com.datadog.android.rum.internal.monitor.NoOpRumMonitor
+import java.util.UUID
+
+interface RumMonitor {
+
+    fun addUserAction(
+        action: String,
+        attributes: Map<String, Any?> = emptyMap()
+    )
+
+    fun startView(
+        key: Any,
+        className: String,
+        attributes: Map<String, Any?> = emptyMap()
+    )
+
+    fun stopView(
+        key: Any,
+        className: String,
+        attributes: Map<String, Any?> = emptyMap()
+    )
+
+    fun addResource(
+        contentType: String,
+        url: String,
+        duration: Long,
+        attributes: Map<String, Any?> = emptyMap()
+    )
+
+    fun activeContext(): RumContext
+
+    // region Builder
+
+    /**
+     * A Builder class for a [RumMonitor].
+     */
+    class Builder(private val applicationId: UUID) {
```

Comment:
why do you want to expose the applicationId ? You don't want to generate this internally in the constructor ?

Reply:
The application_id is provided by the developer, like the token, so we need it.